### PR TITLE
Add max on collection, collection-proxy, and test.

### PR DESCRIPTION
**Pull request by dragonstin**
Add max method, from issue #15.
Adjusted my previous commit to including changes in the collection-proxy and test files.

Please, feel free to request any changes.

### CODE REVIEW

File: src/collection-proxy.js [hunk 1]
```diff
@@ -361,6 +361,17 @@ class CollectionProxy {
...
   }
 
+  /**
+   * Returns the max value in the collection.
+   *
+   * @returns {Integer}
+   */
+  max () {
+    return this.all(
+      this._enqueue('size')
```
Comment (marcuspoehls) [inline src/collection-proxy.js]: Please enqueue the `max` method name.

```suggestion
      this._enqueue('max')
```

**Some background on the internals:**

The collection proxy allows chainable calls to run multiple functions in series. Collecting all methods in a call chain is required to process async callbacks. At some point, the chain needs a `.all()` call to retrieve the result.

The collection proxy will then call the related methods in the collection class. The collection class has the actual implementation, the proxy just ensures the order and any steps necessary to keep the collections immutable.

**Review by marcuspoehls — CHANGES_REQUESTED** [review]
@fernandadagostin Hey Fernanda, thank you for the PR! I’m requesting a typo fix and guess the tests will pass as soon as you change the enqueued method from `size` to `max`. I’ve added a comment on the internals of this lib, they may be helpful to understand why enqueuing the correct method is required :)

File: src/collection.js [hunk 2]
```diff
@@ -382,6 +382,15 @@ class Collection {
     return this.items.length
   }
 
+  /**
+   * Returns the max value in the collection.
+   *
+   * @returns {Number}
+   */
+  max() {
```
Comment (marcuspoehls) [inline src/collection.js]: Can you please a space between before the function parenthesis?

```suggestion
  max () {
```

**Review by marcuspoehls — CHANGES_REQUESTED** [review]
@fernandadagostin I’m requesting one more minor formatting change :)

File: src/collection-proxy.js (hunk 1 above)
Comment (dragonstin) [reply src/collection-proxy.js]: Done! I am sorry for my mistake.

File: src/collection.js (hunk 2 above)
Comment (dragonstin) [reply src/collection.js]: Done.

**Review by marcuspoehls — APPROVED** [review]

**Comment by marcuspoehls** [comment]
@fernandadagostin Thank you for your PR and continuous support throughout the refinements. Merged!

**Comment by dragonstin** [comment]
Thanks for your attention and cooperation.

**Pull request merged**